### PR TITLE
K8SPXC-552 Add secrets volume to haproxy and pxc

### PR DESCRIPTION
**Pull request by nonemax**
[K8SPXC-552](https://jira.percona.com/browse/K8SPXC-552)

### CODE REVIEW

File: pkg/pxc/app/statefulset/haproxy.go [hunk 1]
```diff
@@ -86,12 +86,6 @@ func (c *HAProxy) AppContainer(spec *api.PodSpec, secrets string, cr *api.Percon
 				Name:  "PXC_SERVICE",
 				Value: c.labels["app.kubernetes.io/instance"] + "-" + "pxc",
 			},
-			{
-				Name: "MONITOR_PASSWORD",
```
Comment (delgod) [inline pkg/pxc/app/statefulset/haproxy.go:90]: it should be in place if the cr version less than 1.7.0

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | failed |
| storage | failed |
| haproxy | failed |
| big-data | passed |
| recreate | passed |
| scaling | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| security-context | failed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| init-deploy | failed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| limits | failed |
| monitoring | failed |
| scheduled-backup | passed |
| monitoring-2-0 | failed |
| upgrade-proxysql | failed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| smart-update | failed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-consistency | failed |

File: pkg/pxc/app/statefulset/haproxy.go (hunk 1 above)
Comment (hors) [reply pkg/pxc/app/statefulset/haproxy.go:90]: @nonemax  please fix all tests too

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | failed |
| scaling | passed |
| storage | failed |
| haproxy | failed |
| big-data | passed |
| recreate | passed |
| scaling-proxysql | failed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| security-context | failed |
| demand-backup | passed |
| init-deploy | failed |
| limits | failed |
| self-healing-advanced | passed |
| monitoring | failed |
| operator-self-healing | passed |
| monitoring-2-0 | failed |
| upgrade-proxysql | failed |
| smart-update | failed |
| affinity | failed |
| one-pod | passed |
| scheduled-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-cert-manager | passed |
| upgrade-consistency | failed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

**Review by delgod — DISMISSED** [review]

**Review by hors — CHANGES_REQUESTED** [review]
@nonemax please fix the tests

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| haproxy | failed |
| big-data | passed |
| upgrade-haproxy | failed |
| recreate | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| security-context | failed |
| restore-to-encrypted-cluster | failed |
| init-deploy | passed |
| upgrade-proxysql | failed |
| self-healing-advanced | passed |
| demand-backup | failed |
| smart-update | passed |
| limits | passed |
| monitoring | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| monitoring-2-0 | failed |
| affinity | passed |
| one-pod | failed |
| upgrade-consistency | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

File: Jenkinsfile [hunk 2]
```diff
@@ -123,6 +123,7 @@ pipeline {
         VERSION = "${env.GIT_BRANCH}-${env.GIT_SHORT_COMMIT}"
         CLUSTER_NAME = sh(script: "echo jenkins-pxc-${GIT_SHORT_COMMIT} | tr '[:upper:]' '[:lower:]'", , returnStdout: true).trim()
         AUTHOR_NAME  = sh(script: "echo ${CHANGE_AUTHOR_EMAIL} | awk -F'@' '{print \$1}'", , returnStdout: true).trim()
+        IMAGE_HAPROXY="perconalab/percona-xtradb-cluster-operator:k8spxc-552-haproxy"
```
Comment (delgod) [inline Jenkinsfile:126]: please revert before merge

**Review by delgod — APPROVED** [review]

**Review by hors — CHANGES_REQUESTED** [review]
@nonemax what about users test? As we discussed this test should use not only proxysql but haproxy too. Please improve it to avoid such type of issue in the future.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| storage | passed |
| haproxy | passed |
| scaling | passed |
| upgrade-haproxy | failed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| demand-backup | passed |
| monitoring-2-0 | failed |
| smart-update | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | passed |
| validation-hook | passed |
| upgrade-consistency | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| storage | passed |
| upgrade-haproxy | passed |
| haproxy | passed |
| big-data | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| security-context | failed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| limits | passed |
| monitoring-2-0 | passed |
| smart-update | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| haproxy | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| limits | passed |
| security-context | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| upgrade-haproxy | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| storage | passed |
| validation-hook | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| haproxy | passed |
| storage | passed |
| scaling | passed |
| big-data | passed |
| recreate | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| smart-update | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | failed |
| users | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-691-3a9581bd

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| haproxy | passed |
| big-data | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | passed |
| self-healing-advanced | passed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | passed |
| validation-hook | passed |

File: e2e-tests/users/run [hunk 3]
```diff
@@ -81,4 +81,13 @@ kubectl_bin apply \
...
 compare_mysql_cmd "select-4" "SHOW TABLES;" "-h $cluster-proxysql -uoperator -p$pass"
 
+newpass="test-password2"
+newpassencrypted=$(echo -n "$newpass" | base64)
+spinup_pxc "$cluster" "$test_dir/conf/some-name.yml" 3 10
+wait_cluster_consistency "$cluster" 3 3
+patch_secret "my-cluster-secrets" "monitor" "$newpassencrypted"
+sleep 15
+wait_cluster_consistency "$cluster" 3 3
+compare_mysql_cmd "select-3" "SHOW DATABASES;" "-h $cluster-haproxy -umonitor -p$newpass"
```
Comment (delgod) [inline e2e-tests/users/run:91]: what the difference between this test and test in 44-48 lines?

**Review by delgod — CHANGES_REQUESTED** [review]

File: e2e-tests/users/run (hunk 3 above)
Comment (nonemax) [reply e2e-tests/users/run:91]: we check haproxy here

Comment (delgod) [reply e2e-tests/users/run:91]: thank you!

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| haproxy | passed |
| upgrade-haproxy | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| smart-update | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| demand-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | failed |
| scheduled-backup | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Review by blez — DISMISSED** [review]

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-691-f39bb2be

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| big-data | passed |
| scaling | passed |
| scaling-proxysql | passed |
| storage | passed |
| security-context | passed |
| self-healing | passed |
| recreate | passed |
| self-healing-advanced | passed |
| restore-to-encrypted-cluster | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

File: Jenkinsfile (hunk 2 above)
Comment (delgod) [reply Jenkinsfile:127]: https://github.com/percona/percona-docker/pull/378 is merged, please remove the line

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

**Review by blez — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| storage | passed |
| haproxy | failed |
| big-data | passed |
| recreate | passed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| init-deploy | passed |
| limits | passed |
| upgrade-proxysql | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| smart-update | passed |
| one-pod | passed |
| scheduled-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-691-85f81012

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| haproxy | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| upgrade-proxysql | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| smart-update | passed |
| upgrade-consistency | passed |

**Pull request merged**